dnf5-plugins/needs_restarting_plugin/needs_restarting.cpp

```diff
@@ -277,7 +299,18 @@ void NeedsRestartingCommand::services_need_restarting(Context & ctx) {
         const auto start_timestamp_us =
             uint64_t{unit_proxy->getProperty("ActiveEnterTimestamp").onInterface(SYSTEMD_UNIT_INTERFACE)};
 
-        unit_file_to_service.insert(std::make_pair(fragment_path, Service{unit_name, start_timestamp_us}));
+        services.push_back(SystemdService{unit_name, start_timestamp_us, unit_object_path, fragment_path});
```

Comment:
~~I think you should be able to use `emplace_back` here too, the type is clear:~~
```suggestion
        services.emplace_back(unit_name, start_timestamp_us, unit_object_path, fragment_path);
```

_edit_: never mind, there's no constructor for it…